### ISSUE TITLE
Component theme should have proper typings

### ISSUE THREAD

**Issue by trajano** [GeekyAnts/NativeBase#4144]
Looking at the code for Button it just uses `any` for the prop.  Ideally there should be a type.  Something like this but I don't know if it can be applied to all the scenarios that you may have.
```ts
import type { ITheme } from "native-base";
import type { ColorSchemeName } from "react-native";
import type { StylesProps } from "styled-system";

type StyleFunctionBaseProps = {
  theme: ITheme;
  colorMode: NonNullable<ColorSchemeName>;
  accessibleColors: boolean;
  setColorMode(colorMode: NonNullable<ColorSchemeName>): void;
  setAccessibleColors(accessibleColors: boolean): void;
};
type StyleProps<P> = { [styleKey: string]: any } & Partial<P>;
type StyleFunction<P> = (props: P & StyleFunctionBaseProps) => StylesProps;
export type ComponentTheme<
  P extends Record<string, unknown> = { [propKey: string]: any }
> = {
  baseStyle?: StyleFunction<P> | StyleProps<P>;
  defaultProps?: P;
  sizes?: Record<string, StyleProps<P>>;
  variants?: Record<string, StyleFunction<P> | StyleProps<P>>;
};

```

**Comment by trajano** [comment]
With that you can ensure type safety with this example

```ts
const cardTheme: ComponentTheme = {
  baseStyle: (props) => {
    console.log(props.accessibleColors);
    return { round: "md" };
  },
  defaultProps: {
    variant: "image",
  },
};
export default cardTheme;
```

**Comment by makkarMeenu** [comment]
Fixed in NativeBase 3.2.2 . Let us know if you still have issues.
